src/component/MovieModal/MovieModal.ts

```diff
@@ -0,0 +1,296 @@
+      myVoteButtonWrapperElement.addEventListener('click', (event) => {
+        const targetElement = event.target as HTMLElement;
+        if (!targetElement.classList.contains('my-vote-button-wrapper')) {
+          const myVoteButtonWrapperElement = targetElement.closest(
+            '.my-vote-button-wrapper',
+          ) as HTMLDivElement;
+          const myVoteButtonElement = targetElement.closest(
+            '.my-vote-button',
+          ) as HTMLButtonElement;
+
+          const voteData = {
+            id: Number(myVoteButtonWrapperElement.id),
+            score: Number(myVoteButtonElement.value),
+          };
+
+          addUserVotesToLocalStorage(voteData);
+          this.#updateMyVote(voteData.id);
+        }
+      });
+    }
+  }
+
+  setMovieModalItem(movieDetail: IMovieDetailData) {
+    const titleElement = $('.modal-title') as HTMLSpanElement;
+    const posterElement = $('.modal-poster') as HTMLImageElement;
+    const genresElement = $('.modal-genres') as HTMLSpanElement;
+    const voteAverageElement = $('.modal-vote-average') as HTMLSpanElement;
+    const overviewElement = $('.modal-overview') as HTMLParagraphElement;
+    const myVoteButtonWrapperElement = $('.my-vote-button-wrapper');
+
+    if (myVoteButtonWrapperElement) {
+      myVoteButtonWrapperElement.id = String(movieDetail.id);
+      this.#updateMyVote(movieDetail.id);
+    }
+
+    titleElement.textContent =
+      movieDetail.title || '해당 영화는 제목 정보를 제공하지 않습니다.';
+
+    posterElement.src = movieDetail.poster_path
+      ? `${IMAGE_BASE_URL}${movieDetail.poster_path}`
+      : posterEmptyImg;
+
+    const names = movieDetail.genres.map((genre: IGenre) => genre.name);
+    genresElement.textContent =
+      movieDetail.genres.length !== 0
+        ? names.join(', ')
+        : '해당 영화는 장르 정보를 제공하지 않습니다.';
+
+    const voteAverage = movieDetail.vote_average;
+    voteAverageElement.textContent = voteAverage
+      ? String(voteAverage.toFixed(1))
+      : '해당 영화는 평점 정보를 제공하지 않습니다.';
+
+    const overview = movieDetail.overview;
+    overviewElement.textContent = overview
+      ? overview
+      : '해당 영화는 줄거리 정보를 제공하지 않습니다.';
+  }
+
+  #updateMyVote(movieID: number) {
+    const myVoteButtonWrapperElement = $('.my-vote-button-wrapper');
+    const myVoteScoreNumberElement = $('.my-vote-score-number');
+    const myVoteScoreTextElement = $('.my-vote-score-text');
+    const voteScore = getUserVoteScoreFromLocalStorage(movieID);
+    const starCount = voteScore / 2 - 1;
+
+    if (myVoteButtonWrapperElement) {
+      myVoteButtonWrapperElement.childNodes.forEach((myVoteButton, index) => {
+        const myVoteButtonElement = myVoteButton as HTMLButtonElement;
+        const myVoteButtonImageElement =
+          myVoteButtonElement.firstChild as HTMLImageElement;
+        if (index <= starCount) {
+          if (myVoteButtonImageElement) {
+            myVoteButtonImageElement.src = starFilledImage;
+          }
+        } else {
```

Comment:
이렇게 인덴트를 줄여볼 수 있을 것 같아요.
```suggestion
        if (index <= starCount && myVoteButtonImageElement) {
              myVoteButtonImageElement.src = starFilledImage;
        } else {
```